src/avro_binary_decoder.erl

```diff
@@ -134,27 +149,33 @@ dec(Bin, T, _Lkup, Hook) when ?IS_FIXED_TYPE(T) ->
        end).
 
 %% @private
--spec dec_record(binary(), record_type(), lkup_fun(), hook()) ->
-        {avro:out(), binary()}.
-dec_record(Bin, T, Lkup, Hook) ->
+-spec dec_record(binary(), record_type(), lkup_fun(),
+                 hook(), decoder_options()) -> {avro:out(), binary()}.
+dec_record(Bin, T, Lkup, Hook, #{record_type := RecordType} = Options) ->
   FieldTypes = avro_record:get_all_field_types(T),
   {FieldValuesReversed, Tail} =
     lists:foldl(
       fun({FieldName, FieldType}, {Values, BinIn}) ->
-        {Value, BinOut} = dec_item(T, FieldName, FieldType, BinIn, Lkup, Hook),
+        {Value, BinOut} = dec_item(T, FieldName, FieldType,
+                                   BinIn, Lkup, Hook, Options),
         {[{FieldName, Value} | Values], BinOut}
       end, {[], Bin}, FieldTypes),
-  FieldValues = lists:reverse(FieldValuesReversed),
-  {FieldValues, Tail}.
+  FieldValues0 = lists:reverse(FieldValuesReversed),
```

Comment:
no need to reverse for `maps:from_list/1`. i.e. put this to the `proplist` clause.